src/wayland/mod.rs

```diff
@@ -340,20 +340,23 @@ impl WaylandHelper {
         };
         self.inner.conn.flush().unwrap();
 
-        let buffer_infos = session
+        let Some(buffer_infos) = session
             .wait_while(|data| data.buffer_infos.is_none())
             .buffer_infos
             .take()
-            .unwrap();
```

Comment:
Or actually, I think this `unwrap` should never fail since it occurs after `.wait_while(|data| data.buffer_infos.is_none())`? So it should be statically imposible for this to fail and a check is unnecessary, I believe.

Unless you saw this specific unwrap fail?

Reply:
No, I only saw the other one fail. I just didn't notice that this one couldn't ever fail and added removed the unwrap anyway. I'll undo that change.